module/vision/VisualMesh/src/visualmesh/VisualMeshRunner.cpp

```diff
@@ -50,6 +53,28 @@ namespace module::vision::visualmesh {
 
     namespace generate_runner {
 
+        template <template <typename> class Engine, typename Scalar>
+        struct BuildEngine;
+
+        template <typename Scalar>
+        struct BuildEngine<::visualmesh::engine::opencl::Engine, Scalar> {
+            static std::shared_ptr<::visualmesh::engine::opencl::Engine<Scalar>> build(
+                ::visualmesh::NetworkStructure<float> net,
+                std::string cache) {
+                fs::create_directory(cache);  // create the cache folder if it doesn't exist
```

Comment:
You probably want to make all of them if you decide to nest your directory beyond 1 level
```suggestion
                fs::create_directories(cache);  // create the cache folder if it doesn't exist
```